src/pgduckdb_hooks.cpp

```diff
@@ -185,7 +185,7 @@ IsAllowedStatement(Query *query, bool throw_error) {
 	}
 
 	/* We don't support modifying statements on Postgres tables yet */
-	if (query->commandType != CMD_SELECT) {
+	if (ContainsFromClause(query) && query->commandType != CMD_SELECT) {
```

Comment:
This seems a a weird place to add this check. It seem to have anything to do with checking whether the postgres table is a modifying statement or not.

Reply:
I agree, though the line below uses `list_nth_node` with `query->rtable` which could be `NULL` without this check, thus failing the `Assert(list != NIL)` in `list_nth_cell`

Reply:
Though re-reading now with fresh eyes it was indeed pretty weird. Reworked here: https://github.com/duckdb/pg_duckdb/pull/757/commits/2ed8ee441ddd4a0e74bb184f5872dc6235baca00